lib/Parse/ParseDecl.cpp

```diff
@@ -3887,6 +3887,16 @@ Parser::parseDecl(ParseDeclOptions Flags,
     if (Flags.contains(PD_HasContainerType) &&
         IsAtStartOfLineOrPreviousHadSemi) {
 
+      // Fail fast if we have something that starts with an identifier
+      // but cannot be a var or func declaration.
+      const bool NotAVarOrFuncDecl = Tok.isIdentifierOrUnderscore() &&
+        peekToken().isAny(tok::period);
+
+      if (NotAVarOrFuncDecl) {
+          diagnose(Tok, diag::expected_decl);
+          break;
+      }
```

Comment:
Some suggestions:

1. Please rename `NotAVarOrFuncDecl` to `IsProbablyStmt`.
2. We can get rid of `found_unexpected_stmt` because it is not being used.
3. Add a comment to the test case indicating what is being tested. Say something like `Check that we avoid emitting func/var diagnostics when we see something statement-like`.